tests/typecheck/managers/querysets/test_annotate.yml

```diff
@@ -0,0 +1,207 @@
+-   case: annotate_basic
+    main: |
+        from myapp.models import User
+        from django.db.models.expressions import F
+
+        qs = User.objects.annotate(foo=F('id'))
+        reveal_type(qs) # N: Revealed type is 'django.db.models.manager.Manager[myapp.models.User (annotated)]'
```

Comment:
Quick question: can we use the annotation type in the type's repr? Would it be more or less readable / useful?

Reply:
Not sure what you mean by "the annotation type". Is that `foo`? It would be cool to know what it was annotated with, but that's outside the scope of this PR.
Anyway, from my experimentation, we cannot put anything with `.` in the type's name or it will interfere with lookups.